common/js/src/messaging/common.js

```diff
@@ -56,4 +56,22 @@ function nonFatalDefaultServiceCallback(channel, serviceName, payload) {
   channel.dispose();
 }
 
+/**
+ * Helper function which extracts key from message (first it makes sure the key
+ * is actually present).
+ * @param {!Object|string} message
+ * @param {string} key
+ */
+GSC.MessagingCommon.extractKey = function(message, key) {
+  GSC.Logging.checkWithLogger(LOGGER, goog.isObject(message));
+  goog.asserts.assert(goog.isObject(message));
+  GSC.Logging.checkWithLogger(
+      LOGGER,
+      goog.object.containsKey(message, key),
+      'Key "' + key + '" not present in message ' +
+      GSC.DebugDump.dump(message));
```

Comment:
It's safer to use debugDump instead of dump here, because GSC.MessagingCommon.extractKey is a generic function that may be called with a security/privacy-sensitive data.